File: frontend/honeylandAngular/src/app/data/proizvod.ts
import { Pcelar } from './user';

export class Proizvod {
    id: number;

    info: string;

    naziv: string;

    cena: number;

    kolicina: number;

    slika: string;

    kategorija: string;

    slikaShow: any;

    ocena: number;

    brOcena: number;

    pcelar: Pcelar;
}

export class ProizvodPost {
    pcelar: string;

    info: string;

    naziv: string;

    cena: number;

    kolicina: number;

    slika: string;

    kategorija: string;

    ocena: number;

    brOcena: number;
}

export class Porudzbina{
    id:number
    proizvodId: number;
    userId: number;
    kolicina: number;
    datum: Date
    prizvod: Proizvod
}

File: backend/demo/src/main/java/com/example/demo/entity/Proizvod.java
package com.example.demo.entity;

import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.Data;

@Entity
@Table(name = "proizvod")
@Data
public class Proizvod {
	
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
	private Long id;
	
	private String info;
	
	private String naziv;
	
	private int cena;
	
	private int kolicina;
	
	private String slika;
	
	private String kategorija;
	
	
	private BigDecimal ocena;
	
	private int brOcena;
	
	@ManyToOne
	@JoinColumn(name = "pcelar_id")
	private Pcelar pcelar;

}
